### ISSUE TITLE
Redis - 'topics.regex' property required when 'topics' is set

### ISSUE THREAD

**Issue by Kanris826** [Landoop/kafka-connect-ui#80]
Kafka-connect version is `1.1.1-cp1`.
I run kafka-connect-ui from source. Thats Redis connector configuration:

```
name=RedisSinkConnector
connector.class=com.datamountaineer.streamreactor.connect.redis.sink.RedisSinkConnector
topics=TopicName_RedisSinkConnector
tasks.max=1
connect.redis.host=redis-1.m1.example.cloud
connect.redis.port=6379
connect.redis.kcql=INSERT INTO TABLE1 SELECT * FROM redis-topic
```

UI requires `topics.regex` to be set, when `topics` and `topics.regex` are mutually exclusive options.

![image](https://user-images.githubusercontent.com/18378985/42814312-81e52470-89cc-11e8-9237-6d1c107b68b0.png)

But it works fine when kafka-connect version is `1.0.0-cp1 `

**Comment by bingzhan0613** [comment]
When will the problem be solved?

**Comment by andmarios** [comment]
Hi @bingzhan0613, thanks for the ping. We are looking into it, but I can't give you a date yet.

The problem is that the error comes from connect itself (really poor validation here), so we have to intercept and change somehow your settings, in order to pass connect validation. :(

**Comment by ultimatebiggi** [comment]
@Kanris826 Last night we encounter same issue while configuring influx sink connector in our docker environment. So for temporary work around, we used old version of kafka-connect to configure our connector configuration and once that configuration is complete we updated to latest version again. 

Here are the steps we took to make our life easy. Below is snippet from docker compose file. I hope that will help you to resolve your problem for now. 
 
Do configuration using below version 
  connect:
    image: confluentinc/cp-kafka-connect:3.3.0
Once configuration is done 
  connect:
    image: confluentinc/cp-kafka-connect:5.0.0

**Comment by jglambed** [comment]
@Kanris826 @bingzhan0613 This is fixed with the 0.9.6 release. I will close the issue. 
Thank you for your feedback

**Comment by thirstyfish** [comment]
@jglambed I'm still experiencing this validation issue with the Cassandra Sink when setting up the connector from Connect UI.

I tried with the following Confluent connect versions
confluentinc/cp-kafka-connect:4.1.2 (uses Kafka 1.1)
confluentinc/cp-kafka-connect:5.0.1 (uses Kafka 2.0)

I tried both images with the connector for Kafka version 1.0 and 1.1 to no avail, still getting the validation error in Landoop's Connect UI:

_Config "topics.regex" requires a value
Required config "topics.regex" is not there_

**Comment by andmarios** [comment]
Pull the latest docker or rebuild from master, then refresh your browser (for chrome, open the developer console, then hit F5).

**Comment by thirstyfish** [comment]
Thanks @andmarios. It seems there's still an issue with the syntax check for the Cassandra sinks.
A setting like this
```
connect.cassandra.kcql=INSERT INTO events SELECT EVENT_ID as eventId FROM CSAS_CASSANDRA TTL=7776000
```
Throws an _Invalid syntax_ error, because of the = sign in the KCQL query.

Is there a workaround for this or should I create a new issue for this one?

Cheers

**Comment by andmarios** [comment]
Hi @thirstyfish.

I believe the issue you experience was fixed with commit 590c6a24226dad9fd13037e6aefcc2fdc8fbb5f7.
There isn't a release yet that includes this commit, but you can build your own for now.

**Comment by andmarios** [comment]
Hi @thirstyfish, took some time but the release is out: 0.9.7.
If you use the docker image, you can use the tags `latest` and `0.9.7`. :)

**Comment by thirstyfish** [comment]
> Hi @thirstyfish, took some time but the release is out: 0.9.7.
> If you use the docker image, you can use the tags `latest` and `0.9.7`. :)

Great stuff, thanks @andmarios , this has resolved the KCQL TTL issue. Cheers!
